src/proxy/inbound.rs

```diff
@@ -302,30 +309,30 @@ impl Inbound {
         };
         let ds =
             proxy::guess_inbound_service(&rbac_ctx.conn, &for_host, upstream_service, &upstream);
-        let connection_metrics = ConnectionOpen {
-            reporter: Reporter::destination,
-            source,
-            derived_source: Some(derived_source),
-            destination: Some(upstream),
-            connection_security_policy: metrics::SecurityPolicy::mutual_tls,
-            destination_service: ds,
-        };
-        let result_tracker = metrics::ConnectionResult::new(
+        let result_tracker = Arc::new(metrics::ConnectionResult::new(
```

Comment:
Can we just construct + wrap the instantiated struct with `Arc` from inside `ConnectionResult::new`?

Reply:
yep, will change

Reply:
NVM, we use a Box in other codepaths but require atomic here. IMO its best to let the callers decide